### PR TITLE
Small cleanup to make setDoorPosition more DRY

### PR DESCRIPTION
**Pull request by bdraco**

### CODE REVIEW

File: components/ratgdo/ratgdo.cpp [hunk 1]
```diff
@@ -675,39 +676,36 @@ namespace ratgdo {
...
-            set_timeout("move_to_position", closing_time, [=] {
-                doorCommand(data::DOOR_STOP);
-                this->movingToPosition = false;
-                this->doorPosition = position;
-            });
-            this->positionSyncWhileClosing(delta);
+            this->positionSyncWhileClosing(-delta);
         }
+
+        auto operation_time = duration * 1000 * delta;
```
Comment (bdraco) [inline components/ratgdo/ratgdo.cpp:699]: I think on close `set_timeout` could have been called with a negative value before?

```
            delta = -delta;
            auto closing_time = this->closingDuration * 1000 * delta;
```

Comment (mariusmuja) [reply components/ratgdo/ratgdo.cpp:699]: Before it was in the `if (delta<0)` condition body, but in this PR seems like you're not accounting for that.

File: components/ratgdo/ratgdo.cpp [hunk 2]
```diff
@@ -675,39 +676,36 @@ namespace ratgdo {
...
+        } else { // close
             doorCommand(data::DOOR_CLOSE);
-            this->movingToPosition = true;
-            set_timeout("move_to_position", closing_time, [=] {
-                doorCommand(data::DOOR_STOP);
-                this->movingToPosition = false;
-                this->doorPosition = position;
-            });
-            this->positionSyncWhileClosing(delta);
+            this->positionSyncWhileClosing(-delta);
```
Comment (bdraco) [inline components/ratgdo/ratgdo.cpp]: delta has to < 0 here because if its == 0 we already returned above

**Comment by mariusmuja** [comment]
Looks good now!

**Review by mariusmuja — APPROVED** [review]

**Pull request merged**